rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+
+function Example() {
+  const shakeAnimation = useSharedValue(0); // Create a shared value to hold the value of shakeAnimation
+
+  const shakeAnimationStyle = useAnimatedStyle(() => { // Define a worklet that is triggered whenever shakeAnimation changes
+    return {
+      transform: [
+        {
+          translateX: interpolate( // Interpolate the animated value in the range of 0 to 100 to create a shake effect
+            shakeAnimation.value,
+            [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100], // Input range
+            [0, -50, 50, -50, 50, -50, 50, -50, 50, -50, 0], // Output range
+          ),
+        },
+      ],
+    };
+  });
+
+  return (
+    <View>
+      <Animated.View style={[styles.block, shakeAnimationStyle]} />
+      <Button text="Shake" onPress={() => {
+        shakeAnimation.value = withTiming(100, { // Change the value of shakeAnimation to 100
+          duration: motionToken.duration.duration3, // Define the duration it takes to change the value of shakeAnimation to 100
+          easing: motionToken.easing.standard.effective, // Define the easing of the animation
+        });
+      }}
+      />
+    </View>
+  );
+}
+```
+
+- React Native's **Animated API** works on the **same principle as React Native**; you define your **animations** **in JS** and they **execute** on the **Native** **platforms**.
+- The **downside** of this approach is that it still needs to **communicate with** the Native realm using **the bridge** which can lead to **dropped frames** in complex animations in cases where the **bridge is already choked up**.
+- **React native reanimated** (v2) aims to provide ways of **offloading animation** and event handling logic off of the JavaScript thread and onto the **UI thread**.
+- This is achieved by defining **Reanimated worklets** – chunks of JavaScript code that can be moved to a **separate JavaScript VM** and executed synchronously on the UI thread. This makes it possible to respond to touch events immediately and update the UI within the same frame when the event happens **without worrying about** the load that is put on the **main JavaScript thread**. 
+- In the above example, we are using `useAnimatedStyle` to create a worklet that will be executed on the UI thread and has a **shared value** `translateXOffset` that is shared by JS as well as native realms.
+- React Native reanimated is able to achieve this using React Native's `TurboModules` feature.
+- Since **we want to leverage these benefits** of **React Native Reanimated**, **we will be using it** for our native **animations on Blade**.
+
+
+# Drawbacks/Constraints
+### Library specific tokens
+Consumers would need to be aware that certain tokens are library specific and would not work with other libraries
+- For web, we expose `duration` as a `string` of *milliseconds* which works well with CSS's `transition-duration` property but will not work with Framer motion since it requires a `number` in *seconds*
```

Comment:
We might want to use framer or react spring at some point in the future. We'd have to relook at the tokens then. For those use cases, we could have helpers in place and we can be okay but for use cases where we'd wanted to use `Easing.bezier` from react native's `Animated` API instead of Easing.bezier` of react native reanimated, we might have to make a good amount of refactor since we're planning to expose the token as a function of `Easing.bezier(...)` where `Easing` is imported from a specific library.